test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -26,6 +26,69 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
       assert_not_authorised
     end
 
+    should "raise an exception if the user doesn't have access to the application" do
```

Comment:
I wonder if it's worth grouping this and the access token test into a context, since they could both be considered "user doesn't have access" tests, and then making sure in this test that they do have an access token, just not the signin permission